src/nodes.coffee

```diff
@@ -753,7 +762,11 @@ exports.SuperCall = class SuperCall extends Call
   # method.
   superReference: (o) ->
     method = o.scope.namedMethod()
-    if method?.klass
+    if method?.isMethod and method?.ctor
```

Comment:
`superReference` is in a bit of a state now. In order to preserve behaviours such as `super` in indirect methods and soaked super invocations, the old implementation hasn't been removed. If those were removed this would become much, much simpler.